src/main/resources/application.properties

```diff
@@ -0,0 +1,8 @@
+spring.datasource.url=jdbc:postgresql://localhost:5432/postgres
+spring.datasource.username=postgres
+spring.datasource.password=redert77
+spring.jpa.hibernate.ddl-auto=update
+spring.jpa.show-sql=true
+spring.security.user.name=admin
+spring.security.user.password=geeksforgeeks
+spring.main.allow-circular-references=true
```

Comment:
Why do you need circular reference?

Reply:
I don't, I realize its bad, I placed it there as a workaround because my code kept running into a circular reference error, 
I managed to fix it by placing @Lazy on my user service but I came to realize that's just a smaller workaround, now at the end I resolved it by injecting the user service straight into the security config which I'm still unsure of being the correct way of doing it, but it seems the least wrong out of all of them.